File: 1.kt
fun main() {
    val S = readLine()!! // Исходная строка для сжатия
    var compressed = "" // Строка для хранения сжатого результата
    var count = 1 // Счётчик повторяющихся символов
    
    // Проходим по строке, если символ совпадает с предыдущим, увеличиваем счётчик
    for (i in 1 until S.length) {
        if (S[i] == S[i - 1]) {
            count++
        }
        // Если символы разные, добавляем предыдущий символ и счётчик в результат
        else {
            compressed += S[i - 1]
            compressed += count.toString()
            count = 1
        }
    }
    
    // Добавляем последний символ и его количество
    compressed += S.last()
    compressed += count.toString()
    
    // Проверяем, эффективно ли сжатие
    if (compressed.length < S.length) {
        println(compressed) // Выводим сжатую версию
    } else {
        println(S) // Выводим исходную строку
    }
}

File: 1.rs
use std::io;

fn main() {
    let mut S = String::new();  // Исходная строка для сжатия
    io::stdin().read_line(&mut S).expect("Ошибка чтения строки");
    let S = S.trim();
    let mut compressed = String::new();  // Строка для хранения сжатого результата
    let mut count = 1;      // Счётчик повторяющихся символов
    let chars: Vec<char> = S.chars().collect();
    
    // Проходим по строке, если символ совпадает с предыдущим, увеличиваем счётчик
    for i in 1..chars.len() {
        if chars[i] == chars[i - 1] {
            count += 1;
        }
        // Если символы разные, добавляем предыдущий символ и счётчик в результат
        else {
            compressed.push(chars[i - 1]);
            compressed.push_str(&count.to_string());
            count = 1;
        }
    }
    
    compressed.push(chars[chars.len() - 1]);
    compressed.push_str(&count.to_string());
    
    // Проверяем, эффективно ли сжатие
    if compressed.len() < S.len() {
        println!("{}", compressed);  // Выводим сжатую версию
    } else {
        println!("{}", S);  // Выводим исходную строку
    }
}
